### ISSUE TITLE
var-spacing: false-positive in regexp

### ISSUE THREAD

**Issue by skwde** [ansible/ansible-lint#2285]
##### Summary

Rather similar to <https://github.com/ansible/ansible-lint/issues/2208> but with variables.

The pipe in the `regexp: '({{ var1 }}|{{ var2 }})'` triggers a false-positive.

Also related

- <https://github.com/ansible/ansible-lint/issues/2241>
- <https://github.com/ansible/ansible-lint/issues/2209>

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
# ansible --version
ansible [core 2.13.2]
  config file = $HOME/ansible-test/ansible.cfg
  configured module search path = ['$HOME/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/miniconda3/envs/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = $HOME/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/miniconda3/envs/ansible/bin/ansible
  python version = 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0]
  jinja version = 3.1.2
  libyaml = True

# ansible-lint --version
ansible-lint 6.4.0 using ansible 2.13.2
```

- ansible installation method: conda
- ansible-lint installation method: conda

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behavior

The regexp should not trigger a var-spacing warning

##### Actual Behavior

A minimal playbook might look like

```yml
---
- name: test
  hosts: localhost
  gather_facts: false
  tasks:
    - name: test
      ansible.builtin.lineinfile:
        path: /some/file.txt
        regexp: '({{ var1 }}|{{ var2 }})'
        line: 'abc'
```

The pipe triggers a var-spacing warning.
